SDWebImage/SDMemoryLRUCache.m

```diff
@@ -0,0 +1,495 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+
+#import "SDMemoryLRUCache.h"
+#import "SDImageCacheConfig.h"
+#import "SDInternalMacros.h"
+#import "UIImage+MemoryCacheCost.h"
+#import <CoreFoundation/CoreFoundation.h>
+#import <pthread.h>
+
+static void *SDMemoryLRUCacheContext = &SDMemoryLRUCacheContext;
+
+static inline dispatch_queue_t SDMemoryCacheGetReleaseQueue() {
+    return dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_LOW, 0);
+}
+/**
+ * A node in deque map.
+ */
+@interface SDMemoryCacheMapNode : NSObject {
+    @package
+    __unsafe_unretained SDMemoryCacheMapNode *_pre;
```

Comment:
Xcode 10 supports `ARC Objective-C object inside the struct filed`. See [WWDC2018](https://developer.apple.com/videos/play/wwdc2018/409/). Should we have a plan to remove the Xcode 9 support in the future (5.x or 6.x)